tests/shopify_python/test_packaging.py

```diff
@@ -0,0 +1,102 @@
+import contextlib
+import re
+import subprocess
+
+from git import repo
+import pytest
+
+from shopify_python import packaging
+
+
+@pytest.fixture
+def package_source_root(tmpdir):
+    # type: ('py.path.LocalPath') -> 'py.path.LocalPath'
+    root = tmpdir.join('test_packaging')
+
+    root.join('setup.py').write('\n'.join([
+        "import setuptools",
+        "setuptools.setup(",
+        "    name='test_packaging',",
+        "    install_requires=[",
+        "        'shopify_python'",
+        "    ],",
+        "    entry_points={",
+        "        'egg_info.writers': [",
+        "            'git_sha.txt = shopify_python.packaging:write_package_revision',",
+        "        ],",
+        "    }",
+        ")"]),
+        ensure=True
+    )
+    return root
+
+@pytest.fixture
+def git_package_root(package_source_root):
+    # type: ('py.path.LocalPath') -> 'py.path.LocalPath'
+    package_repo = repo.Repo.init(str(package_source_root))
+    package_repo.index.add([str('setup.py')])
+    package_repo.index.commit("initial commit")
+    return package_source_root
+
+@pytest.fixture
+def package_root_with_revision_file(package_source_root):
+    # type: ('py.path.LocalPath') -> 'py.path.LocalPath'
+    package_source_root.join('REVISION').write("0080d5ea79fa9513cd7a1b5aeef19858cf52430f")
+    return package_source_root
+
+@contextlib.contextmanager
+def package_installed(path, develop_mode):
+    # type: ('py.path.LocalPath', bool) -> None
+
+    # sdist is what initializes the egg-info.
+    try:
+        subprocess.check_output(['python', 'setup.py', 'sdist'], cwd=str(path))
+    except subprocess.CalledProcessError as exc:
+        print exc.output
+        raise
+
+    base_command = ['pip', 'install']
+    flags = ['-e'] if develop_mode else []
+    arguments = [str(path)]
+
+    try:
+        subprocess.check_output(base_command + flags + arguments)
+    except subprocess.CalledProcessError as exc:
+        print exc.output
+        raise
+
+    yield
+
+    try:
+        subprocess.check_output(['pip', 'uninstall', '-y', 'test_packaging'])
+    except subprocess.CalledProcessError as exc:
+        print exc.output
+        raise
+
+# We can't just call shopify_python.packaging.get_package_revision within the current interpreter because pkg_resources
+# initializes its DB of package info when first invoked and then doesn't update (after, for example, we install the
+# test package).
+def call_get_package_revision():
+    # type: () -> str
+    return subprocess.check_output(['python', '-c', '\n'.join([
+        "import shopify_python.packaging",
+        "print(shopify_python.packaging.get_package_revision('test-packaging'))"
+    ])]).decode().strip()
+
+@pytest.mark.parametrize('develop_mode', [True, False])
+def test_git_package(develop_mode, git_package_root):
+    with package_installed(git_package_root, develop_mode=develop_mode):
+        assert re.match('^[a-zA-Z0-9]{40}$', call_get_package_revision()) is not None
```

Comment:
Should you not also assert that the revision is not `0080d5ea79fa9513cd7a1b5aeef19858cf52430f`?